src/docfx/build/xref/XrefMap.cs

```diff
@@ -94,9 +94,17 @@ public IEnumerable<XrefSpec> InternalReferences
             Dictionary<string, XrefSpec> map = new Dictionary<string, XrefSpec>();
             foreach (var url in docset.Config.Xref)
             {
-                var json = File.ReadAllText(docset.GetFileRestorePath(url));
-                var xRefMap = JsonUtility.Deserialize<XrefMapModel>(json);
-                foreach (var spec in xRefMap.References)
+                var content = File.ReadAllText(docset.GetFileRestorePath(url));
+                XrefMapModel xrefMap = new XrefMapModel();
+                if (url.EndsWith(".yml", StringComparison.OrdinalIgnoreCase))
+                {
+                    xrefMap = YamlUtility.Deserialize<XrefMapModel>(content);
+                }
+                else if (url.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+                {
+                    xrefMap = JsonUtility.Deserialize<XrefMapModel>(content);
+                }
```

Comment:
log error if neither JSON nor YAML?

Reply:
Seems like we are not logging an error if file extension not matched?
https://github.com/dotnet/docfx/blob/ff26cee570ccbcb528321a4c564d2b4061a49c77/src/docfx/config/Config.cs#L255

Reply:
The caller has a guard to handle this:
https://github.com/dotnet/docfx/blob/ff26cee570ccbcb528321a4c564d2b4061a49c77/src/docfx/config/Config.cs#L174

Reply:
I think that's because we always need a config file, but we are not warning the user that some extension are not supported.
Like this one as well, we do not warn the user the page extension is not supported, just make sure it is what we want:
https://github.com/dotnet/docfx/blob/45a0c9c7951c3bb9c7a7be98ce6b1a2b449c0b58/src/docfx/build/page/BuildPage.cs#L87